### ISSUE TITLE
Switch to pre-commit.ci for pre-commit on CI.

### ISSUE THREAD

**Issue by Swiftb0y** [mixxxdj/mixxx#11306]
pre-commit.ci is the CI service run by the maintainer of pre-commit. It would make sense to transition to them for two primary reasons: more efficient resource use (which also results in faster builds) and on-request automatic fixing which would significantly lower the barrier to entry for casual contributors (see https://github.com/mixxxdj/mixxx/pull/11279#issuecomment-1444333724). 

Thoughts?

**Comment by ronso0** [comment]
Sounds fun! The **speed** graph is impressive (though pre-commit is not the bottleneck, is it?)

**Comment by Swiftb0y** [comment]
Not really, though from what I can tell each pre-commit run currently always has to install its dependencies first which is rather wasteful in terms of bandwidth and CPU time. The speed is not the primary reason for switching though. It is rather to have one-click solution for the PRs where a contributor refuses or doesn't manage to install pre-commit where we can just take care of it for them (without us having to checkout the branch and run pre-commit ourselves each time).
